bindings/java/src/main/java/ethereum/ckzg4844/CKZG4844JNI.java

```diff
@@ -106,6 +105,31 @@ public static int getBytesPerBlob() {
    */
   public static native void loadTrustedSetup(byte[] g1, long g1Count, byte[] g2, long g2Count);
 
+  /**
+   * An alternative to {@link #loadTrustedSetup(String)}. Loads the trusted setup from a resource.
+   *
+   * @param resource the resource name that contains the trusted setup
+   * @param clazz    the class to use to get the resource
+   * @throws CKZGException if there is a crypto error
+   * @throws IllegalArgumentException if the resource does not exist
+   */
+  public static void loadTrustedSetupFromResource(String resource, Class clazz) {
+    InputStream is = clazz.getResourceAsStream(resource);
+    if (is == null) {
+      throw new IllegalArgumentException("Resource " + resource + " does not exist.");
+    }
+
+    try (is) {
+      Path jniWillLoadFrom = tempDir().resolve("kzg-trusted-setup.txt");
+      jniWillLoadFrom.toFile().deleteOnExit();
+      Files.copy(is, jniWillLoadFrom, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
This feels like there could be a race condition. Is it possible to create a unique temp directory each time `tempDir()` is called, instead of reusing the same one? I guess it could be a limitation of `deleteOnExit`.

Reply:
testing on Linux I had no issues, in any case I have reverted the change, so now it just create a temp file in the default tmp dir, instead of trying to reuse the same dir of the lib

Reply:
Thanks, personally I like this better 👍